src/parse/core/cache.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include <nlohmann/json.hpp>
+#include "state.hpp"
+#include "symbol.hpp"
+
+namespace docgen {
+namespace parse {
+namespace core {
+
+// Cache is a structure that needs to be remembered
+// across batches and between routines.
+// MaxSymbolSize is the maximum symbol length (not including null-terminator).
+template <size_t MaxSymbolSize>
+struct Cache
```

Comment:
I don't really see why this needs to exist as a template struct... but besides that, it is poorly named. I think we should name our classes/structs by functionality (or if really just a packing struct like this, name by contents), not their intended usage. Also, this has nothing to do with my understanding of what a "cache" is xD

Reply:
Because `Symbol` requires it... I disagree with your naming. If you can come up with a better name let me know. When you cache something, you store it for near-future use. Is this not a good name?

Reply:
A cache is a where information that was recently retrieved is stored temporarily in case it needs to be quickly accessed again. It looks like this is state information which will be pertinent throughout the duration we are parsing a given document. They are unrelated things.

Reply:
`quickly accessible` and `storing temporarily` properties are a consequence of how we create cache in CPU. A cache is first and foremost simply a storage of things that will later be used in the future. If you have a better name, please suggest.

Reply:
I would call it `ParseStatus`, or maybe just `Status` given the nature of the information being stored. Idrc though, this was kind of a silly thing for me to push on across multiple comments xD Although I don't see why this has to be a template still. Couldn't you just take the symbol size in the constructor?

Reply:
Shit ok `Status` is pretty fucking good xD. Let me change that haha.

The reason why I wanted it to be a template param was because `std::array` requires the size as a template parameter and all template paramters must be known at compile-time. Passing size into constructor is a run-time known value and therefore cannot be passed as size of `std::array` for Symbol.

But atm, I'm actually fine with changing this such that we have a `std::vector` now that I think about it. We can take size from `Symbol` constructor, and use this to reserve the size of `vector` object. Then it's like a fixed array except it'll just be on the heap. We'll benchmark later if we need to further optimize by using arrays, but we def got bigger probs to worry about atm.